sympy/solvers/ode/tests/ode_test_suit/test_kamke.py

```diff
@@ -0,0 +1,112 @@
+"""
+File to test Kamke ODEs. To run test suite, run
+
+    python test_kamke.py
+
+Information about failing ODEs is logged in test_report.txt.
+"""
+
+
+from contextlib import contextmanager
+import threading
+import _thread
+import time
+
+from sympy import symbols, Function, Eq, dsolve, sympify, checkodesol
+
+
+class TimeOutError(Exception):
+    def __init__(self, msg=''):
+        self.msg = msg
+
+
+@contextmanager
+def time_limit(seconds, msg=''):
+    timer = threading.Timer(seconds, lambda: _thread.interrupt_main())
+    timer.start()
+    try:
+        yield
+    except KeyboardInterrupt:
+        raise TimeOutError("Timed out for operation {}".format(msg))
+    finally:
+        # Cancel timer if process finishes in time
+        timer.cancel()
+
+
+start = time.time()
+# Read all parsed ODEs from text file
+f = open("kamke_parsed.txt", "r")
+eqs = f.readlines()
+f.close()
+# Open a file to log failing ODEs
+f = open("test_report.txt", "w")
```

Comment:
Reading from a file should use `with`:
```
with open(filename) as input_file:
    lines = input_file.readlines()
```
This kind of thing should not be done at top-level in the module (it should be in a function).

Reply:
Yes, this was just a very basic draft to get it running.